src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -706,6 +719,15 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return emit_simd_ins_for_sig (cfg, klass, OP_CREATE_SCALAR, -1, arg0_type, fsig, args);
 	case SN_CreateScalarUnsafe:
 		return emit_simd_ins_for_sig (cfg, klass, OP_CREATE_SCALAR_UNSAFE, -1, arg0_type, fsig, args);
+	case SN_Equals:
+		return emit_xcompare (cfg, klass, arg0_type, args [0], args [1]);
+	case SN_EqualsAll:
+		return emit_xequal (cfg, klass, args [0], args [1]);
```

Comment:
What code is `xequal` generating?

I'm just wanting to ensure that this is effectively just `cmpeqps + testz` (and therefore `NaN != NaN`).

This is different from `IEquatable.Equals` where `NaN.Equals(NaN)` must be `true`

Reply:
On x64 this is what I get:

```csharp
private static bool EqualsAll(Vector64<double> left, Vector64<double> right)
    => Vector64.EqualsAll(left, right);

private static void Main(string[] args)
{
    var a = Vector64.Create(1.4);
    var b = Vector64.Create(1.4);

    Console.WriteLine($"EqualsAll({a}, {b}) == {EqualsAll(a, b)}");

    var c = Vector64.Create(Double.NaN);
    var d = Vector64.Create(Double.NaN);

    Console.WriteLine($"EqualsAll({c}, {d}) == {EqualsAll(c, d)}");
}
```

```
0000000000000000 <loWorld_Program_EqualsAll__System_Runtime_Intrinsics_Vector64_1_double__System_Runtime_Intrinsics_Vector64_1_double__>:
<BB>:1
   0:   c4 e1 f9 6e c7          vmovq  %rdi,%xmm0
   5:   c4 e1 f9 6e ce          vmovq  %rsi,%xmm1
   a:   c5 fb c2 c1 00          vcmpeqsd %xmm1,%xmm0,%xmm0
   f:   c4 e1 f9 7e c0          vmovq  %xmm0,%rax
  14:   83 e0 01                and    $0x1,%eax
  17:   c3                      retq
***

EqualsAll(<1.4>, <1.4>) == True
EqualsAll(<NaN>, <NaN>) == False
```

Reply:
hmm but this seems broken (although it hasn't been touched in this PR):
```csharp
private static bool TestEquals(Vector64<double> left, Vector64<double> right)
    => left.Equals(right);

private static void Main(string[] args)
{
    var a = Vector64.Create(Double.NaN);
    var b = Vector64.Create(Double.NaN);

    Console.WriteLine($"{a}.Equals({b}) == {TestEquals(a, b)}");
}
```

```
0000000000000000 <loWorld_Program_TestEquals__System_Runtime_Intrinsics_Vector64_1_double__System_Runtime_Intrinsics_Vector64_1_double__>:
<BB>:1
   0:   c4 e1 f9 6e c7          vmovq  %rdi,%xmm0
   5:   c4 e1 f9 6e ce          vmovq  %rsi,%xmm1
   a:   c5 fb c2 c1 00          vcmpeqsd %xmm1,%xmm0,%xmm0
   f:   c4 e1 f9 7e c0          vmovq  %xmm0,%rax
  14:   83 e0 01                and    $0x1,%eax
  17:   c3                      retq
***
```

Reply:
> hmm but this seems broken (although it hasn't been touched in this PR):

I believe this is an issue with the managed implementation and has a tracking issue for it.

Otherwise looks good, thanks!